Update for eForms

duncandewhurst opened the pull request.

duncandewhurst committed:
Update for eForms

odscjen commented on README.md:
```diff
@@ -26,7 +29,10 @@ In the European Union, this extension's fields correspond to [eForms BG-102 (Sub
         "durationInDays": 74
       },
-      "depositsGuarantees": "An on demand performance bond which value is a percentage of the total contract price issued by an acceptable entity by the Contracting Entity (e.g. bank or insurance company)."
+      "depositsGuarantees": "An on demand performance bond which value is a percentage of the total contract price issued by an acceptable entity by the Contracting Entity (e.g. bank or insurance company).",
```
```suggestion
      "depositsGuarantees": "An on demand performance bond issued by an entity (e.g. bank or insurance company) judged acceptable by the Contracting Entity. The value of the bond is a percentage of the total contract price.",
```
This whole sentence is confusing. "..bond which value..." jars, should it not be "... bond whose value...". And "...issued by an acceptable entity by the Contracting Entity..." is also confused, who is doing the issuing, who is judging acceptable? So suggested an alternative that is hopefully still a correct description but which is also still not great..

duncandewhurst replied:
I went with:

> An on-demand performance bond issued by an entity that the Contracting Entity judges to be acceptable (e.g. a bank or insurance company) and whose value is a percentage of the total contract price.

duncandewhurst committed:
README.md: Copy-edit example

jpmckinney commented on README.md:
```diff
@@ -6,17 +6,20 @@ Adds a submission terms object to the tender and lot objects, to describe how, w
 ## Legal context
 
-In the European Union, this extension's fields correspond to [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/), although not all the fields have been implemented yet. See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
+In the European Union, this extension's fields correspond to the following [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/en/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http ...
```
It says "the following" but nothing follows?

Reusing formulation from other extensions.

```suggestion
In the European Union, this extension's fields correspond to [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). See [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/) for the correspondences to eForms fields. See [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/master/en/) for the correspondences to Tenders Electronic Daily (TED).
```

duncandewhurst replied:
Ah, just a typo. Fixed in https://github.com/open-contracting-extensions/ocds_submissionTerms_extension/pull/4/commits/51ef97b1d291c7bc99d7eb3852ccdd62ee4d95d0 in order to preserve the formulation that we're now using.

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+            "boolean",
+            "null"
+          ],
+          "description": "Whether an advanced or qualified electronic signature or seal (as defined in European Parliament and Council Regulation (EU) No 910/2014) is required."
```
Put description after title.

duncandewhurst replied:
I think I ordered the keywords according to your suggestion in https://github.com/open-contracting/european-union-support/issues/173#issuecomment-1542949533. Is there a reason for different orders in different places?

jpmckinney replied:
Oh, I meant the `type` field in that comment, not the `type` JSON Schema keyword.

duncandewhurst replied:
Ah, I see! Hopefully my misunderstanding didn't propagate too far!

jpmckinney replied:
No worries! I'll correct it across extensions, looks like there's maybe just one new case in the statistics extension, and some pre-existing order differences.

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+        },
+        "subcontractingClauses": {
+          "title": "Subcontracting clauses",
+          "description": "The information about subcontracting that must be indicated in the tender.",
```
Style guide is to use bid. Replace normative word. EU uses "indicate" in an awkward way, so changing.

```suggestion
          "description": "The information about subcontracting that needs to be provided in the bid.",
```

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+        },
+        "nonElectronicRationale": {
+          "title": "Non-electronic submission rationale",
+          "description": "The rationale for electronic submission not being possible.",
```
EU uses "not allowed" in the codelist, but "not possible" in the field description, which is just confusing.

```suggestion
          "description": "The rationale for electronic submission not being allowed.",
```

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+          ],
+          "description": "Whether tenderers can submit more than one bid."
+        },
+        "nonElectronicRationale": {
```
The field this relates to is `electronicSubmissionPolicy`. Let's make that clearer.

Need to update the example and changelog.

```suggestion
        "nonElectronicSubmissionRationale": {
```

duncandewhurst replied:
Example and changelog updated in https://github.com/open-contracting-extensions/ocds_submissionTerms_extension/pull/4/commits/8f7fe4c302dc7296b5714183b741399bffa14cd9

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+            "boolean",
+            "null"
+          ],
+          "description": "Whether an advanced or qualified electronic signature or seal (as defined in European Parliament and Council Regulation (EU) No 910/2014) is required."
```
Let's leave it up to each jurisdiction to decide the legal definition.

```suggestion
          "description": "Whether an advanced or qualified electronic signature or seal is required."
```

duncandewhurst committed:
Apply suggestions from code review

Co-authored-by: James McKinney <26463+jpmckinney@users.noreply.github.com>

duncandewhurst committed:
README.md: Update for nonElectronicSubmissionRationale

duncandewhurst committed:
README.md: Fix typo

duncandewhurst committed:
Merge branch 'master' into eforms

jpmckinney commented on README.md:
```diff
@@ -6,17 +6,20 @@ Adds a submission terms object to the tender and lot objects, to describe how, w
 ## Legal context
 
-In the European Union, this extension's fields correspond to [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/), although not all the fields have been implemented yet. For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/latest/en/).
+In the European Union, this extension's fields correspond to [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/en/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.op ...
```
```suggestion
In the European Union, this extension's fields correspond to [eForms BG-102 (Submission Terms)](https://docs.ted.europa.eu/eforms/latest/reference/business-terms/). For correspondences to eForms fields, see [OCDS for eForms](https://standard.open-contracting.org/profiles/eforms/latest/en/). For correspondences to Tenders Electronic Daily (TED), see [OCDS for the European Union](http://standard.open-contracting.org/profiles/eu/latest/en/).
```

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+        },
+        "advancedElectronicSignatureRequired": {
+          "title": "Advanced electronic signature required",
+          "type": [
+            "boolean",
+            "null"
+          ],
+          "description": "Whether an advanced or qualified electronic signature or seal is required."
```
Also changed required -> needed to avoid RFC keyword.

```suggestion
          "description": "Whether an advanced or qualified electronic signature or seal is needed.",
          "type": [
            "boolean",
            "null"
          ]
```

jpmckinney commented on release-schema.json:
```diff
@@ -98,6 +98,45 @@
+        },
+        "multipleBidsAllowed": {
+          "title": "Multiple bids allowed",
+          "type": [
+            "boolean",
+            "null"
+          ],
+          "description": "Whether tenderers can submit more than one bid."
```
```suggestion
          "description": "Whether tenderers can submit more than one bid.",
          "type": [
            "boolean",
            "null"
          ]
```

jpmckinney committed:
Apply suggestions from code review

release-schema: Use non-normative word when describing the contracting process. Re-order description JSON Schema keyword property. readme: Avoid URL redirect.
